packages/app/src/specs/generators/component/ReactComponentGeneratorStepOne.vue

```diff
@@ -0,0 +1,198 @@
+<template>
+  <div class="flex flex-col flex-grow justify-between">
+    <template v-if="generatedSpecError">
+      <EmptyGenerator
+        :gql="generateSpecFromSource.currentProject"
+        title=""
+        type="component"
+        :other-generators="false"
+        :spec-file-name="generatedSpecError.fileName"
+        @restart="cancelSpecNameCreation"
+        @updateTitle="(value) => emits('update:title', value)"
+      />
+    </template>
+
+    <template v-else>
+      <div class="flex-grow">
+        <div
+          v-if="generateSpecMutation.fetching.value"
+          class="mt-48px w-full inline-flex items-center justify-center"
+        >
+          <i-cy-loading_x16 class="h-48px mr-12px animate-spin w-48px" />
+          <p class="text-lg">
+            Loading
+          </p>
+        </div>
+        <ExpandableFileChooser
+          v-else-if="!result"
+          v-model:extensionPattern="extensionPattern"
+          :files="allFiles"
+          :loading="query.fetching.value"
+          @select-item="makeSpec"
+        />
+        <GeneratorSuccess
+          v-else
+          :file="result.file"
+        />
+      </div>
+      <div>
+        <StandardModalFooter
+          v-if="result"
+          class="flex gap-16px items-center"
+        >
+          <router-link
```

Comment:
The Button component knows how to render a router link if you give it the `to` prop, so this might be able to be simplified to just use Button: https://github.com/cypress-io/cypress/blob/develop/packages/frontend-shared/src/components/Button.vue#L121

(Our designs treat "button" as a purely visual concept, that sometimes acts like a link or a button, so the code treats it the same way... wonder if there's a way to make this more discoverable/obvious, possibly with the component name?)